bodo/libs/_bodo_to_arrow.cpp

```diff
@@ -720,7 +720,7 @@ std::shared_ptr<arrow::Table> bodo_table_to_arrow(
 
 std::shared_ptr<arrow::Table> bodo_table_to_arrow(
     std::shared_ptr<table_info> table, std::vector<std::string> field_names,
-    std::shared_ptr<arrow::KeyValueMetadata> schema_metadata,
+    const std::shared_ptr<const arrow::KeyValueMetadata> schema_metadata,
```

Comment:
```suggestion
    const std::shared_ptr<arrow::KeyValueMetadata> schema_metadata,
```
I think a while ago we found putting const in the shared_ptr type to be really bad for performance, I can try and dig up the doc if you'd like